extra_exercises/reverse_linked_list.py

```diff
@@ -0,0 +1,33 @@
+"""Print a linked list in reverse order. Follow-up: try to do it without additional memory (simple variables are fine)."""
+from linked_list.linked_list import LinkedList
+
+
+def reverse_with_list(linked_list):
+    reverse_linked = []
+    current = linked_list.head
+    while current.next is not None:
+        reverse_linked.insert(0, current.item)
+        current = current.next
+    reverse_linked.insert(0, current.item)
+    return reverse_linked
+
+
+def reverse_with_variables(linked_list):
```

Comment:
you can't do this followup because in order to do it you need a real linked list and not just an array. you developed a linked list some time ago, so i suggest you at some point to retry using the appropriate data structure